File: decoupled/backends/go/api/models/activity_log.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type ActivityLog struct {
	gorm.Model
	UserID    uint
	ActionID  uint
	Timestamp time.Time
}

type Action struct {
	gorm.Model
	Name         string
	ActivityLogs []ActivityLog
}

File: decoupled/backends/python/api/models/activity_log.py
from sqlalchemy import Column, DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import relationship

from models.base_model import BaseModel


class ActivityLog(BaseModel):
    __tablename__ = "activity_log"

    user_id = Column(Integer, ForeignKey("user.id"))
    action_id = Column(Integer, ForeignKey("activity.id"))
    timestamp = Column(DateTime)


class Activity(BaseModel):
    __tablename__ = "activity"

    name = Column(String)
    activity_logs = relationship("ActivityLog", back_populates="activity")
